### DISCUSSION TITLE
How to offer translation help?

### DISCUSSION THREAD

**Discussion by jcfrog** [aclap-dev/video-downloadhelper#110] [Language & Translation]
The add-on now integrates its own translation system and allow anyone to modify almost any string displayed in the add-on.

Furthermore, once you have modified some strings, you can export them to a file. If you post this file in this forum, your changes will be integrated into the next add-on version so those strings will be the default for everyone using the language you specified.

To access the translation system, open VDH settings (for instance, click the VDH toolbar icon, then the gear icon at the bottom right of the panel), then More and Translation. You can now see all the strings used in the interface, along with the original en-US string.

If your browser language is not English (for which all strings are present in the add-on), you will see a dropdown switch at the top right with options Missing strings and All strings. The first options only shows the strings for which there is translated string in your language, resulting in the English string being displayed. You can use the second option to see (and then modify) the strings that have already been  translated.

Once done with your changes, click **Save** then **Export** to get a JSON file to your disk. Come to this [Language & Translation forum](https://github.com/aclap-dev/video-downloadhelper/discussions/categories/language-translation), create a [new topic](https://github.com/aclap-dev/video-downloadhelper/discussions/new?category=language-translation) with subject containing the name of your locale into brackets, for instance _Update to French locale [fr]_, and attach the file to the topic.

In order to get the latest add-on version, it is better you run the latest beta-version from https://www.downloadhelper.net/firefox/betas (just install it once, you will get the updates automatically).

Before doing any translation work, it is wise to check in the last topics of this forum that your target locale has not already been provided by someone else. If you are unsure, please ask in this forum about the status of the locale you want to update. This may prevent you from doing a translation job that has already been done. We've now switched to these Github discussions, but you can still check out the old [Google group](https://groups.google.com/g/video-downloadhelper-internationalization) we used to use.

**Comment by calipo69** [comment]
Hello jcfrog!
I have translated "Update to German locale [de] #360". 
I am new in the group. I am just asking if it is ok or not?
txs for your answer.
best regards kurt

**Reply by jcfrog** [reply]
looks good to me, the devs will be able to use it for next releases. Thanks.

**Comment by AdventureTime111** [comment]
Hey folks,
Is anyone currently working on the Czech localization? If not, I’m attaching the exported file with my translation. However, it would be great if someone sighted could possibly take a quick look. I'm blind and I'm having a bit of trouble with the new UI, making it hard for me to use/find some of the elements such as the blacklist, so unfortunately I was unable to test everything myself.
Many thanks.
[cs.txt](https://github.com/aclap-dev/video-downloadhelper/files/15417488/cs.txt)

**Reply by paulrouget** [reply]
Thank you, I'll check!

**Comment by Anard** [comment]
Hi. Just updated 1 string in french translation : 
[fr.txt](https://github.com/aclap-dev/video-downloadhelper/files/15452092/fr.txt)

**Comment by willy-liu** [comment]
Hello. I filled up all of the zh_TW empty translation slot here.
[zh_TW.txt](https://github.com/user-attachments/files/16374147/zh_TW.txt)

**Comment by TondaK** [comment]
I am adding the missing Czech translations.
[cs.txt](https://github.com/user-attachments/files/16855506/cs.txt)

**Comment by ronnonl** [comment]
Here are the missing translations for The Netherlands.
[nl.txt](https://github.com/user-attachments/files/17535013/nl.txt)

**Comment by roggaru** [comment]
Hello polish translation.
[pl.txt](https://github.com/user-attachments/files/21117513/pl.txt)
